### PR TITLE
split dataset controller and alluxioruntime controller, edit makefile…

### LINKED ISSUE #100: 将dataset controller和alluxioruntime 进行拆分和解耦
**What feature you'd like to add:**

目前dataset controller和alluxioruntime controller在同一个binary里，需要进行拆分。dataset controller独立负责dataset 声明周期，与实现无关

1.dataset controller 和alluxioruntime controller拆分，分别由两个main函数作为入口
2.两个controller分别部署在不同的Pod

/cc @xieydd @TrafalgarZZZ @iluoeli 

**Why is this feature needed:**

### PR DESCRIPTION
**Pull request by yangyuliufeng**
### Ⅰ. Describe what this PR does
This PR split the runtime_controller into dataset_controller and alluxioruntime_controller.
I have edited the makefile and dockerfile about buinding images for dataset_controller and alluxioruntime_controller.
In my last PR, I printed version log by "glog" package when csi starting in version.go. I habe changed it to "log" package because it seems like that glog.infof will conflict with setupLog in controller.
I have reduce the authority of dataset_controller and alluxioruntime_controller. They used the role of cluster-admin before.
### Ⅱ. Does this pull request fix one issue?

fixes #100

### Ⅲ. List the added test cases (unit test/integration test) if any, please explain if no tests are needed.
I have tested it by "make docker-build-all" and testing part of cases.

### Ⅴ. Special notes for reviews
The image address in chart v4 need to be edited.

### CODE REVIEW

File: Dockerfile.dataset [hunk 1]
```diff
@@ -28,9 +23,8 @@ RUN curl -o /usr/local/bin/kubectl https://storage.googleapis.com/kubernetes-rel
...
 add charts/ /charts
 
-COPY --from=builder /go/bin/fluid-controller /usr/local/bin/fluid-controller
+COPY --from=builder /go/bin/dataset-controller /usr/local/bin/dataset-controller
 #COPY --from=builder /go/bin/dlv /usr/local/bin/dlv
 RUN chmod -R u+x /usr/local/bin/
-CMD fluid-controller
 
-# CMD ["dlv", "--listen=:12345", "exec", "/usr/local/bin/fluid-controller", "--", "--enable-leader-election"]
+CMD dataset-controller start
```
Comment (cheyang) [inline Dockerfile.dataset:30]: 加一个回车符

File: cmd/alluxioruntime_controller/main.go [hunk 2]
```diff
@@ -81,30 +111,14 @@ func main() {
 		Scheme:             scheme,
 		MetricsBindAddress: metricsAddr,
 		LeaderElection:     enableLeaderElection,
-		// MaxConcurrentReconciles: 5,
+		LeaderElectionID:   "7857424864.data.fluid.io",
```
Comment (cheyang) [inline cmd/alluxioruntime_controller/main.go:114]: 为什么要加leaderElectionID？另外MaxConcurrentReconciles删除的原因是什么呢？

File: charts/fluid/v0.4.0/values.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+# Default values for fluid.
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+
+controller:
```
Comment (cheyang) [inline charts/fluid/v0.4.0/values.yaml:5]: 名字改成下面这样如何？

```
dataset:
    controller:
```

**Review by cheyang — CHANGES_REQUESTED** [review]

File: cmd/dataset_controller/main.go [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+			encCfg.EncodeTime = zapcore.ISO8601TimeEncoder
+			o.Encoder = zapcore.NewConsoleEncoder(encCfg)
+		}
+	}))
+
+	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+		Scheme:             scheme,
+		MetricsBindAddress: metricsAddr,
+		LeaderElection:     enableLeaderElection,
+		LeaderElectionID:   "89759796.data.fluid.io",
```
Comment (cheyang) [inline cmd/dataset_controller/main.go:113]: 需要LeaderElectionID吗？

**Review by cheyang — CHANGES_REQUESTED** [review]
非常感谢，总体很赞！

1.请修复下CI中的问题：https://travis-ci.org/github/fluid-cloudnative/fluid/jobs/738993270
2.v0.4.0的helm chart等发布后再做snapshot，我们还是在chart/fluid/fluid下修改吧
3.有些小问题，我们可以讨论

File: cmd/alluxioruntime_controller/main.go (hunk 2 above)
Comment (yangyuliufeng) [reply cmd/alluxioruntime_controller/main.go:114]: leaderElectionID是选主的时候configMapsResourceLock的分布式锁的名字。manager.Options里并没有这个字段MaxConcurrentReconciles。

Comment (cheyang) [reply cmd/alluxioruntime_controller/main.go:114]: MaxConcurrentReconciles 是在这里定义的，后面我们可以加上 https://github.com/kubernetes-sigs/controller-runtime/issues/964

**Review by cheyang — APPROVED** [review]

**Pull request merged**